### PR TITLE
AZP: Update CUDA and gdrcopy versions

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**

### CODE REVIEW

File: contrib/test_jenkins.sh [hunk 1]
```diff
@@ -27,8 +27,8 @@ source $(dirname $0)/../buildlib/tools/common.sh
 
 WORKSPACE=${WORKSPACE:=$PWD}
 ucx_inst=${WORKSPACE}/install
-CUDA_MODULE="dev/cuda11.4"
-GDRCOPY_MODULE="dev/gdrcopy2.3_cuda11.4"
+CUDA_MODULE="dev/cuda12.2.2"
+GDRCOPY_MODULE="dev/gdrcopy2.3.1-1_cuda12.2.2"
```
Comment (yosefe) [inline contrib/test_jenkins.sh:31]: maybe can remove, since test_jenkins.sh sources common.sh

Comment (Alexey-Rivkin) [reply contrib/test_jenkins.sh]: Good idea, done.

**Review by yosefe — APPROVED** [review]

File: buildlib/az-helpers.sh [hunk 2]
```diff
@@ -176,8 +176,8 @@ try_load_cuda_env() {
     if [ -f "/proc/driver/nvidia/version" ]; then
         have_cuda=yes
         have_gdrcopy=yes
-        az_module_load dev/cuda11.4 || have_cuda=no
-        az_module_load dev/gdrcopy2.3_cuda11.4 || have_gdrcopy=no
+        az_module_load dev/cuda12.2.2 || have_cuda=no
```
Comment (artemry-nv) [inline buildlib/az-helpers.sh:179]: Why don't we use latest and greatest CUDA 12.3.1?

File: buildlib/tools/common.sh [hunk 3]
```diff
@@ -4,8 +4,8 @@ WORKSPACE=${WORKSPACE:=$PWD}
 # build in local directory which goes away when docker exits
 ucx_build_dir=$HOME/${BUILD_ID}/build
 ucx_inst=$ucx_build_dir/install
-CUDA_MODULE="dev/cuda11.4"
-GDRCOPY_MODULE="dev/gdrcopy2.3_cuda11.4"
+CUDA_MODULE="dev/cuda12.2.2"
+GDRCOPY_MODULE="dev/gdrcopy2.3.1-1_cuda12.2.2"
```
Comment (artemry-nv) [inline buildlib/tools/common.sh:8]: Should we migrate to/add GDRCOPY 2.4.1?

File: buildlib/az-helpers.sh (hunk 2 above)
Comment (Alexey-Rivkin) [reply buildlib/az-helpers.sh:179]: The version was specifically requested by the customer.

File: buildlib/tools/common.sh (hunk 3 above)
Comment (yosefe) [reply buildlib/tools/common.sh:8]: Seems latest gdrcopy/cuda are these:
```
$ module show dev/gdrcopy-latest
-------------------------------------------------------------------
/hpc/local/etc/modulefiles/dev/gdrcopy-latest:

module-whatis    add gdrcopy to your environment
prepend-path     CPATH /hpc/local/oss/gdrcopy2.3.1-1_cuda12.2.2/include
```

**Pull request merged**